### ISSUE TITLE
Several errors on start task

### ISSUE THREAD

**Issue by andrea-spotsoftware** [ng-select/ng-select#258]
Please fill below information if issuing a bug report. 
### Expected behaviour
Following guidelines for contributing results in some build error

### Actual behaviour
I've just cloned the repository master branch, then after launching yarn start:

`
 [413] ./node_modules/zone.js/dist/long-stack-trace-zone.js 6.22 kB {1} [built]
    + 399 hidden modules

ERROR in [at-loader] ./demo/app/app.component.ts:12:15
    TS2304: Cannot find name 'require'.

ERROR in [at-loader] ./demo/app/layout/header.component.ts:2:22
    TS2304: Cannot find name 'require'.

ERROR in [at-loader] ./demo/app/layout/header.component.ts:3:23
    TS2304: Cannot find name 'require'.

ERROR in [at-loader] ./demo/polyfills.ts:4:1
    TS2304: Cannot find name 'require'.

ERROR in [at-loader] ./demo/polyfills.ts:8:5
    TS2304: Cannot find name 'process'.

ERROR in [at-loader] ./demo/polyfills.ts:15:5
    TS2304: Cannot find name 'require'.

ERROR in [at-loader] ./src/ng-select/ng-select.component.ts:49:15
    TS2304: Cannot find name 'require'.

ERROR in [at-loader] ./src/ng-select/ng-select.component.ts:50:14
    TS2304: Cannot find name 'require'.
Child html-webpack-plugin for "index.html":
         Asset     Size  Chunks  Chunk Names
    index.html  1.46 MB       0
       [0] ./node_modules/html-webpack-plugin/lib/loader.js!./demo/index.ejs 1.58 kB {0} [built]
       [1] ./node_modules/lodash/lodash.js 540 kB {0} [built]
       [2] (webpack)/buildin/global.js 488 bytes {0} [built]
       [3] (webpack)/buildin/module.js 495 bytes {0} [built]
webpack: Failed to compile.
`

### More Info
ng-select version: latest master version clone

browser: -

reproducible in demo page: -

Maybe I am missing something, but I can't figure out what.

**Comment by varnastadeus** [comment]
@andrea-spotsoftware you are definitely missing something. Try reinstalling you node_modules or similar

**Comment by varnastadeus** [comment]
@andrea-spotsoftware `@types` folder should be resolved automatically. Maybe you have old ts version or something

**Comment by nicu-chiciuc** [comment]
I think the issue shouldn't be closes since the instructions on the readme file are not correct or, at least, don't provide all the requirements for successfully building the project. I'm getting the following errors:
```
 + 399 hidden modules

ERROR in [at-loader] ./demo/app/app.component.ts:12:15
    TS2304: Cannot find name 'require'.

ERROR in [at-loader] ./demo/app/layout/header.component.ts:2:22
    TS2304: Cannot find name 'require'.

ERROR in [at-loader] ./demo/app/layout/header.component.ts:3:23
    TS2304: Cannot find name 'require'.

ERROR in [at-loader] ./demo/polyfills.ts:4:1
    TS2304: Cannot find name 'require'.

ERROR in [at-loader] ./demo/polyfills.ts:8:5
    TS2304: Cannot find name 'process'.

ERROR in [at-loader] ./demo/polyfills.ts:15:5
    TS2304: Cannot find name 'require'.

ERROR in [at-loader] ./src/ng-select/ng-select.component.ts:49:15
    TS2304: Cannot find name 'require'.

ERROR in [at-loader] ./src/ng-select/ng-select.component.ts:50:14
    TS2304: Cannot find name 'require'.
Child html-webpack-plugin for "index.html":
         Asset     Size  Chunks  Chunk Names
    index.html  1.46 MB       0
       [0] ./node_modules/html-webpack-plugin/lib/loader.js!./demo/index.ejs 1.53 kB {0} [built]
       [1] ./node_modules/lodash/lodash.js 540 kB {0} [built]
       [2] (webpack)/buildin/global.js 488 bytes {0} [built]
       [3] (webpack)/buildin/module.js 495 bytes {0} [built]
webpack: Failed to compile.
```

Through some digging around I managed to make some of the errors disappear yet I think this shouldn't've happened in the first place.

**Comment by varnastadeus** [comment]
Looks like you have same issues as author of this issue. Have you tried solution he provided?

**Comment by anjmao** [comment]
@andrea-spotsoftware You are correct, windows requires `"typeRoots": [ "../node_modules/@types" ]`
